pkg/controller/elasticsearch/driver/desired_nodes_test.go

```diff
@@ -0,0 +1,809 @@
+					messages: []string{"Successfully calculated compute and storage resources from Elasticsearch resource generation "},
+				},
+			},
+		},
+		{
+			name: "Multi data path",
+			args: args{
+				esReachable: true,
+			},
+			esBuilder: newEs("8.3.0").
+				withNodeSet(
+					nodeSet("default", 3).
+						withCPU("2", "4").
+						withMemory("2Gi", "2Gi").
+						withStorage("1Gi", "1Gi").
+						withNodeCfg(map[string]interface{}{
+							"path.data": []string{"/usr/share/elasticsearch/data1", "/usr/share/elasticsearch/data2"},
+						}),
+				),
+			want: want{
+				result:       wantResult{},
+				deleteCalled: true,
+				condition: &wantCondition{
+					status:   corev1.ConditionFalse,
+					messages: []string{"Elasticsearch path.data must be a string, multiple paths is not supported"},
+				},
+			},
+		},
+		{
+			name: "Elasticsearch client returned an error",
+			args: args{
+				esReachable:   true,
+				esClientError: true,
+			},
+			esBuilder: newEs("8.3.0").
+				withNodeSet(
+					nodeSet("default", 3).
+						withCPU("2", "4").
+						withMemory("2Gi", "2Gi").
+						withStorage("1Gi", "1Gi").
+						pvcCreated(true).
+						withNodeCfg(map[string]interface{}{
+							"path.data": "/usr/share/elasticsearch/data",
+						}),
+				),
+			want: want{
+				result: wantResult{
+					error:  true,
+					reason: "elasticsearch client failed",
+				},
+				condition: &wantCondition{
+					status:   corev1.ConditionTrue,
+					messages: []string{"Successfully calculated compute and storage resources from Elasticsearch resource generation "},
+				},
+			},
+		},
+		{
+			name: "Kubernetes client returned an error",
+			args: args{
+				esReachable:    true,
+				k8sClientError: true,
+			},
+			esBuilder: newEs("8.3.0").
+				withNodeSet(
+					nodeSet("default", 3).
+						withCPU("2", "4").
+						withMemory("2Gi", "2Gi").
+						withStorage("1Gi", "1Gi").
+						pvcCreated(true).
+						withNodeCfg(map[string]interface{}{
+							"path.data": "/usr/share/elasticsearch/data",
+						}),
+				),
+			want: want{
+				result: wantResult{
+					error:  true,
+					reason: "k8s client failed",
+				},
+				condition: &wantCondition{
+					status:   corev1.ConditionUnknown,
+					messages: []string{"Error while calculating compute and storage resources from Elasticsearch resource generation "},
+				},
+			},
+		},
+		{
+			name: "0 values are not allowed",
+			args: args{
+				esReachable: true,
+			},
+			esBuilder: newEs("8.3.0").
+				withNodeSet(
+					nodeSet("default", 3).
+						withCPU("0", "").
+						withMemory("0", "0").
+						withStorage("1Gi", "1Gi").
+						withNodeCfg(map[string]interface{}{
+							"path.data": "/usr/share/elasticsearch/data",
+						}),
+				),
+			want: want{
+				result:       wantResult{},
+				deleteCalled: true,
+				condition: &wantCondition{
+					status: corev1.ConditionFalse,
+					messages: []string{
+						"CPU request is set but value is 0",
+						"Memory limit is set but value is 0",
```

Comment:
Sorry, I missed this one that goes hand in hand.
```suggestion
						"memory limit is set but value is 0",
```